querydsl-jpa/src/main/java/com/querydsl/jpa/EclipseLinkHandler.java

```diff
@@ -54,20 +55,40 @@ public boolean createNativeQueryTyped() {
     @SuppressWarnings("unchecked")
     @Override
     public <T> CloseableIterator<T> iterate(Query query, FactoryExpression<?> projection) {
+        boolean canUseCursor = false;
+        try {
+            canUseCursor = query.unwrap(Query.class) instanceof JpaQuery;
+        } catch (PersistenceException e) { } // can't unwrap, just ignore the exception
```

Comment:
Then this is just your else branch, so you can eliminate the if / else block.

Reply:
It's been 2 years but IIRC I did it this way as a defense against possible buggy wrappers. The Javadoc for javax.persistence.Query is clear that if the requested class can't be unwrapped, then unwrap() should throw a PersistenceException. But there's no way to guarantee a wrapper will implement that correctly - it could just return null instead. Still want me to change it?

Reply:
Actually I recently stumbled upon buggy wrappers as well, so lets keep it like this.